### DISCUSSION TITLE
Memory Leak?

### DISCUSSION THREAD

**Discussion by agillesp** [datalust/seq-tickets#1695] [Help]
The memory usage looks to me like a memory leak. Is this concerning?

<img width="808" alt="Screenshot 2022-11-22 150052" src="https://user-images.githubusercontent.com/1098950/203438646-1b7e38fb-bc4a-46c8-8afa-07abf3b719e1.png">

This is running an 8-core / 60G ECS instance. The service receives 10-50 logs per second.

**Comment by nblumhardt** [comment]
Hi Abe, thanks for reaching out.

This looks fine - Seq will use as much memory as it can for caching, and only start releasing it once system memory utilization hits around 90%.

If you do run into any OOM events or other problems then please let us know, of course! :-)

Best regards,
Nick

**Reply by agillesp** [reply]
Nice. Thanks for the help!

**Comment by agillesp** [comment]
I think we do have a problem. Our Seq has some pretty "ugly" mem usage patterns. And it's indeed crashing regularly. Any suggestions? Thanks.

<img width="653" alt="Screenshot 2022-12-08 123443" src="https://user-images.githubusercontent.com/1098950/206588012-861958f9-8cf4-41b0-b290-4f6f63fe0f60.png">

**Comment by liammclennan** [comment]
Hi @agillesp ,

Please send a diagnostic report from Settings -> Diagnostics and a screenshot of Data -> Storage to support@datalust.co.

**Reply by agillesp** [reply]
This was a few minutes within another crash and recovery.

<img width="973" alt="Screenshot 2022-12-09 071506" src="https://user-images.githubusercontent.com/1098950/206733770-df46ee63-99ec-4c20-b809-d821cd930a1a.png">

I misspoke above, the container instance has 40G, not 60G.

**Reply by agillesp** [reply]
Shoot, just saw you wanted it emailed. I'll send it to the email as well.
